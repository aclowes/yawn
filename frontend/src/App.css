/* Global */

/* WorkflowDetailGraph */

svg.taskGraph {
    height: 500px;
    width: 100%;
    margin: 0 auto;
    display: block;
}

g.node rect {
    stroke: #333;
    fill: #fff
}

g.edgePath path {
    stroke: #333;
    fill: #333;
    stroke-width: 1px;
}