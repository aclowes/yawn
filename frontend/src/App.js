import React from 'react';
import {Navbar, Nav} from 'react-bootstrap';  //NavDropdown, MenuItem

import 'bootswatch/united/bootstrap.css';
import './App.css';
import {YawnNavItem} from './utilities';

class App extends React.Component {

  render() {
    return (
      <div>
        <Navbar>
          <Navbar.Header>
            <Navbar.Brand>YAWN</Navbar.Brand>
          </Navbar.Header>
          <Nav>
            <YawnNavItem href="/">Workflows</YawnNavItem>
            <YawnNavItem href="/workers">Workers</YawnNavItem>
            <YawnNavItem href="/executions">Executions</YawnNavItem>
            <YawnNavItem href="/queues">Queues</YawnNavItem>
            {/*<NavDropdown eventKey={3} title="Dropdown" id="basic-nav-dropdown">*/}
            {/*<MenuItem eventKey={3.1}>Action</MenuItem>*/}
            {/*<MenuItem eventKey={3.2}>Another action</MenuItem>*/}
            {/*<MenuItem eventKey={3.3}>Something else here</MenuItem>*/}
            {/*<MenuItem divider/>*/}
            {/*<MenuItem eventKey={3.3}>Separated link</MenuItem>*/}
            {/*</NavDropdown>*/}
          </Nav>
        </Navbar>
        <div className="container">
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default App;
