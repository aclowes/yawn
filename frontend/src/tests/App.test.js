import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import App from '../App';

import {mockAPI} from './mocks'

it('homepage renders', () => {
  mockAPI([{'username': 'test user'}]);
  const component = ReactTestRenderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls logout on click', () => {
  const mock = jest.fn();
  mock.preventDefault = jest.fn();
  const app = new App();
  app.props = {router: {push: jest.fn()}};
  app.logout(mock);
  expect(mock.preventDefault).toBeCalled();
  expect(app.props.router.push).toBeCalledWith('/login');
});
