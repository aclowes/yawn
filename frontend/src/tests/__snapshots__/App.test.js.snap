// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage renders 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <span
          className="navbar-brand"
        >
          YAWN
        </span>
      </div>
      <div>
        <ul
          className="nav navbar-nav"
          role={null}
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="/workflows"
              onClick={[Function]}
            >
              Workflows
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="/executions"
              onClick={[Function]}
            >
              Executions
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="/workers"
              onClick={[Function]}
            >
              Workers
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="/queues"
              onClick={[Function]}
            >
              Queues
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="/users"
              onClick={[Function]}
            >
              Users
            </a>
          </li>
        </ul>
        <ul
          className="nav navbar-nav navbar-right"
          role={null}
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              <span
                className="glyphicon glyphicon-refresh"
              />
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              role="button"
            >
              Logout 
              (test user)
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  />
</div>
`;
